source/MaterialXGenShader/Util.cpp

```diff
@@ -957,4 +957,40 @@ bool connectsToNormalMapNode(OutputPtr output)
     return connectedNode && connectedNode->getCategory() == "normalmap";
 }
 
+bool hasElementAttributes(OutputPtr output, const StringVec& attributes)
```

Comment:
This is used to detect if any upstream graph from an output may have attributes of interest. In this  case "colorspace" and "unit". Not that we test both inputs and outputs since either could have these attributes applied resulting in additional nodes being added during shader generation which cannot be optimized away.